handlers/abtest/search.go

```diff
@@ -0,0 +1,83 @@
+package abtest
+
+import (
+	"math/rand"
+	"net/http"
+	"time"
+
+	"github.com/ONSdigital/dp-cookies/cookies"
+	"github.com/ONSdigital/log.go/log"
+)
+
+//go:generate moq -out searchtest/handler.go -pkg test . Handler
+type Handler http.Handler
+
+func SearchHandler(newSearch, oldSearch http.Handler, percentage int, domain string) http.Handler {
+	rand.Seed(time.Now().UnixNano())
+
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		// Retrieve AB Test Cookie
+		cookie, err := cookies.GetABTest(req)
+		if err != nil && err != cookies.ErrABTestCookieNotFound {
+			log.Event(req.Context(), "error getting a/b test cookie", log.WARN, log.Error(err))
+		}
+
+		// If AB Test cookie not set, create new AB Test cookie
+		if cookie.NewSearch == nil && cookie.OldSearch == nil {
```

Comment:
Both If statements on line 26 and 39 contain the same code - maybe try to collapse it into a separate function that can be called by both if statements.

Reply:
👍